File: src/mth/nim/ChoiceDialog.java
package mth.nim;

import javafx.fxml.FXML;
import javafx.fxml.FXMLLoader;
import javafx.geometry.HPos;
import javafx.scene.Parent;
import javafx.scene.control.Button;
import javafx.util.Callback;

import java.io.IOException;

public class ChoiceDialog {

    @FXML
    protected Button button1, button2;

    public void setLeftButtonAction(Runnable action) {
        button1.setOnAction(e -> action.run());
    }

    public void setRightButtonAction(Runnable action) {
        button2.setOnAction(e -> action.run());
    }

    public static void showDialog(Callback<HPos, Void> callback) {
        try {
            FXMLLoader fxmlLoader = new FXMLLoader(App.class.getResource("fxml/popup.fxml"));
            Parent content = fxmlLoader.load();
            content.getStylesheets().add("mth/nim/resources/style.css");
            ChoiceDialog controller = fxmlLoader.getController();

            DialogUtil.CustomDialog dialog = new DialogUtil.CustomDialog(content);
            dialog.openDialog();
            controller.setRightButtonAction(() -> {
                dialog.closeDialog();
                callback.call(HPos.RIGHT);
            });
            controller.setLeftButtonAction(() -> {
                dialog.closeDialog();
                callback.call(HPos.LEFT);
            });
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: CoreK/src/mth/nim/ChoiceDialog.kt
package mth.nim

import javafx.event.EventHandler
import javafx.fxml.FXML
import javafx.fxml.FXMLLoader
import javafx.geometry.HPos
import javafx.scene.Parent
import javafx.scene.control.Button
import javafx.util.Callback
import java.io.IOException

open class ChoiceDialog {
    @FXML
    protected lateinit var button1: Button

    @FXML
    protected lateinit var button2: Button

    fun setLeftButtonAction(action: Runnable) {
        button1.onAction = EventHandler { action.run() }
    }

    fun setRightButtonAction(action: Runnable) {
        button2.onAction = EventHandler { action.run() }
    }

    companion object {
        @JvmStatic
        fun showDialog(callback: Callback<HPos, Void>) {
            try {
                val fxmlLoader = FXMLLoader(ChoiceDialog::class.java.getResource("fxml/popup.fxml"))
                val content = fxmlLoader.load<Parent>()
                content.stylesheets.add("mth/nim/resources/style.css")

                val controller = fxmlLoader.getController<ChoiceDialog>()

                val dialog = AnimatedDialog(content)
                dialog.openDialog()

                controller.setRightButtonAction {
                    dialog.closeDialog()
                    callback.call(HPos.RIGHT)
                }
                controller.setLeftButtonAction {
                    dialog.closeDialog()
                    callback.call(HPos.LEFT)
                }
            } catch (e: IOException) {
                e.printStackTrace()
            }
        }
    }
}